bin/fin

```diff
@@ -3085,53 +3119,59 @@ ssh_add ()
 		return 1
 	fi
 
+	# If there are options (e.g., -l), pass arguments as-is and exit
+	if [[ "${1}" =~ ^- ]]; then
+		_ssh_add "$@"
+		return $?
+	fi
+
 	local ssh_path="$HOME/.ssh"
 	# On Windows docker and docker-compose expect a Windows style path
 	is_windows && ssh_path=$(cygpath -w "$ssh_path")
 
 	# Ensure ssh_path exists
 	mkdir -p "$ssh_path"
 
-	if ! is_tty; then
-		# In a non-interactive environment (e.g., CI) run non-interactively. Keys with passphrases cannot be used!
-		echo-yellow "Running in a non-interactive environment. SSH keys with passphrases cannot be used."
-	fi
+	# Copy all keys from the host to the ssh-agent container
+	docker cp "$ssh_path" docksal-ssh-agent:/
 
-	# Only do this when given an argument (key name) or nothing, but not for options (e.g., -l)
-	if [[ ! "${1}" =~ ^- ]]; then
-		# Copy all keys from the host to the ssh-agent container
-		docker cp "$ssh_path" docksal-ssh-agent:/
+	# If we have a key name as the argument, add it to the list of keys to load
+	if [[ "${1}" != "" ]]; then
+		if [[ -f "${ssh_path}/${1}" ]]; then
+			local SECRET_SSH_KEY_${1^^}=${1}
```

Comment:
Getting error message

```
fin: line 3141: SECRET_SSH_KEY_${1^^}=${1}: bad substitution
```

Reply:
Maybe change to something like 

```
                        local ssh_key_name=$(echo $1 | tr '[:lower:]' '[:upper:]')
                        local SECRET_SSH_KEY_${ssh_key_name}=${1}
```

Reply:
I think your bash is not actually bash, by zsh.

I get the same with `zsh`:

```
$ key=server_rsa
$ eval SECRET_SSH_KEY_${key^^}=${key}
zsh: bad substitution
```

With `bash`:

```
bash-4.4$ key=server_rsa
bash-4.4$ eval SECRET_SSH_KEY_${key^^}=${key}
bash-4.4$ echo $SECRET_SSH_KEY_SERVER_RSA
server_rsa
```